### PR TITLE
Adds types and updates to latest go parser

### PR DESCRIPTION
**Pull request by carsonfarmer**
## Summary

Basic updates that add support for typescript types, update to the latest go version, re-compiles the wasm, and reconfigures the file/module layouts. Changes `parse` to `normalize` to better reflect its actual purpose, and to open the door for a real parse function in the future. Also adds support for a customizable max query size, which defaults to the same defaults used in our validator.

## How it was tested

The existing tests continue to pass. Some updates to those tests (to reflect the exported function name change) were required. Additional tests for new functionality around max query size were added, as well as tests to catch common usage patterns that could result in panics. Additionally, this adds testing infrastructure and tests for the manually created types.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by carsonfarmer** [comment]
Hey @awmuncy take a look at this update. It hasn't been released yet, but when I do, I'll bump the minor version. It is technically a breaking change, but I don't think anyone but you is actually using this lib. This changes the `parse` function to `normalize` because that's actually what it does. Plus in the future, I'd like an _actual_ parse function...

Other than that, this adds types which is nice, and a few other cleanups. It also updates to the latest go parser.

**Comment by brunocalza** [comment]
@carsonfarmer In case you missed it, there's a new version released today: https://github.com/tablelandnetwork/go-sqlparser/commit/1b39431a2fea8918ad7d7b7cd6e276ec58e3e8fe

**Comment by carsonfarmer** [comment]
Cheers, I will pull those in!

**Comment by carsonfarmer** [comment]
Updated @brunocalza. I've also added some limited support for maxQuerySize.

File: package.json [hunk 1]
```diff
@@ -4,34 +4,37 @@
   "files": [
```
Comment (awmuncy) [inline package.json:4]: Will you be publishing on npm after this PR goes though? I'm using it in the dashboard through github, but a regular version might be nice. I assume if not previously published, it doesn't _super_ matter, just a thought.

**Review by awmuncy — APPROVED** [review]

File: package.json (hunk 1 above)
Comment (carsonfarmer) [reply package.json:4]: Oh yeh it hasn't been released to npmjs yet. Cool, so I'll likely just release the version as 0.0.1 and then we can go from there. Sound good?

Comment (awmuncy) [reply package.json:4]: Yep, sounds good!

**Pull request merged**